frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -153,60 +127,68 @@ export const Exchange = ({ code, accounts }: TProps) => {
-              <Select
-                options={[{
-                  text: t('buy.exchange.selectRegion'),
-                  value: '',
-                },
-                ...regions]
-                }
-                onChange={(e: React.SyntheticEvent) => setSelectedRegion((e.target as HTMLSelectElement).value)}
-                id="exchangeRegions"
-              />
-            ) : ('')}
-          </div>
-          <div>
-            { !showMoonpay && !showPocket && (
-              t('buy.exchange.noExchanges')
-            )}
-          </div>
-          <div>
-            { showMoonpay && (<Radio
-              disabled={!selectedRegion}
-              id="moonpay"
-              checked={ selectedExchange === 'moonpay' }
-              onChange={() => setSelectedExchange('moonpay')} >
-              {dealsDetails('moonpay')}
-
-            </Radio>) }
-          </div>
-          <div>
-            { showPocket && (<Radio
-              disabled={!selectedRegion}
-              id="pocket"
-              checked={ selectedExchange === 'pocket' }
-              onChange={() => setSelectedExchange('pocket')} >
-              {dealsDetails('pocket')}
-            </Radio>) }
-          </div>
-          <div>
-            <Button
-              primary
-              disabled={!selectedExchange}
-              onClick={goToExchange} >
-          Next
-            </Button>
+          <div className="content narrow isVerticallyCentered">
+            <h1 className={style.title}>{t('buy.title', { name })}</h1>
+            <p className={style.label}>{t('buy.exchange.region')}</p>
+            {regions.length ? <>
+              <div>
+                { regions.length ? (
+                  <Select
+                    options={[{
+                      text: t('buy.exchange.selectRegion'),
+                      value: '',
+                    },
+                    ...regions]
+                    }
+                    onChange={(e: React.SyntheticEvent) => setSelectedRegion((e.target as HTMLSelectElement).value)}
+                    id="exchangeRegions"
+                  />
+                ) : ('')}
+              </div>
+              <div>
+                {noExchangeAvailable && (
+                  <p className={style.noExchangeText}>{t('buy.exchange.noExchanges')}</p>
+                )}
+              </div>
+
+              <div>
+                {!noExchangeAvailable && allExchangeProviders && allExchangeProviders.map(provider => provider.supported && (<ExchangeSelectionRadio
+                  key={provider.providerName}
+                  disabled={!selectedRegion}
+                  id={provider.providerName}
+                  providerName={provider.providerName}
+                  methods={provider.methods}
+                  checked={selectedExchange === provider.providerName}
+                  onChange={() => setSelectedExchange(provider.providerName)} />
+                ))}
+              </div>
+
```

Comment:
With the data already being re-structured and stored into the state `allExchangeProviders`, we can easily render all of them as the custom radio component like this. 

This means, when a new exchage provider is added in the future (which we'll most likely do), there will be a lot less code changes as we don't have to do

`{showMoonPay && moonpay element...}`
`{showPocket && pocket element...}`
`{showSomeOtherNewProvider1 && other provider1...}`
`{showSomeOtherNewProvider2 && other provider2...}`

etc.. 

as it already `.map` automatically.